app/repositories/ReactiveRepositoryHelpers.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import model.Exceptions.{ CannotUpdateRecord, NotFoundException, NotModifiedException, TooManyEntries }
+import play.api.Logger
+import reactivemongo.api.collections.bson.BSONCollection
+import reactivemongo.api.commands.{ UpdateWriteResult, WriteResult }
+import uk.gov.hmrc.mongo.ReactiveRepository
+
+trait ReactiveRepositoryHelpers {
+  this: ReactiveRepository[_, _] =>
+
+  protected lazy val bsonCollection = collection.db.collection[BSONCollection](collection.name)
+
+  def singleUpdateValidator(id: String,
+                            actionDesc: String,
+                            notFound: => Exception): UpdateWriteResult => Unit = {
+    singleUpdateValidatorImpl(id, actionDesc, ignoreNotFound = false, notFound, upsert = false) _
+  }
+
+  def singleUpdateValidator(id: String, actionDesc: String, ignoreNotFound: Boolean = false): UpdateWriteResult => Unit = {
+
+    singleUpdateValidatorImpl(id, actionDesc, ignoreNotFound,
+      new NotFoundException(s"could not find id $id whilst $actionDesc"), upsert = false) _
+  }
+
+  def singleUpsertValidator(id: String,
+                            actionDesc: String): UpdateWriteResult => Unit = {
+
+    singleUpdateValidatorImpl(id, actionDesc, ignoreNotFound = true, new Exception, upsert = true) _
+  }
+
+  def singleRemovalValidator(id: String, actionDesc: String): WriteResult => Unit = (result: WriteResult) => {
+    if (result.ok) {
+      if (result.n == 1) {
+        ()
+      } else if (result.n == 0) {
+        throw new NotFoundException(s"No document found whilst $actionDesc for id $id")
+      } else if (result.n > 1) {
```

Comment:
should we look at using the firstMatchOnly flag on collection.remove call to try and avoid deleting too many documents?